spec/lib/heartcheck/caching_app/cache_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'heartcheck/caching_app/cache'
+require 'concurrent'
+
+module Heartcheck
+  class CachingApp
+    describe Cache do
+      let(:ttl) { 0.1 }
+      let(:controllers) do
+        3.times.collect do |_|
+          Class.new do
+            @count = 0
+
+            def self.call_count
+              @count ||= 0
+            end
+
+            def self.incr
+              @count += 1
+            end
+
+            def index
+              self.class.incr
+              '{"current":"ok"}'
+            end
+          end
+        end
+      end
+
+      def wait_for_one_execution
+        Concurrent::ScheduledTask.execute(ttl) { "waiting the ttl" }.value!
+      end
+
+      subject do
+        described_class.new(controllers, ttl)
+      end
+
+      before do
+        # use a blocking executor, no concurrency
+        subject.concurrent_opts = {executor: Concurrent::ImmediateExecutor.new}
+      end
+
+      describe '#result' do
+        it 'returns an empty result on first dispatch' do
```

Comment:
In this case, it's not `empty`. nil is the correct value for that? In case it's right, just change the text. :)